analyzer/src/main/kotlin/Analyzer.kt

```diff
@@ -166,8 +166,13 @@ class Analyzer(private val config: AnalyzerConfiguration, private val labels: Ma
     }
 
     private fun analyzeInParallel(managedFiles: Map<PackageManager, List<File>>): AnalyzerResult {
-        val state = AnalyzerState()
+        logger.info { "Calling before resolution hooks for ${managedFiles.size} manager(s). " }
+
+        managedFiles.forEach { (manager, definitionFiles) ->
+            manager.beforeResolution(definitionFiles)
```

Comment:
Can you elaborate why you think this is ok with the `mustRunAfter` configuration property? (IIUC, before all lifecycle hooks have been affected by `mustRunAfter`)

Reply:
Can you elaborate why it should not be ok?

Basically, no functionality compared to before is lost: If you need to do work before / after resolution that adheres to `mustRunAfter`, you can easily still do that my moving the code to the begin / end of the `resolveDependencies()` function, as outlined in the commit message.

Reply:
I see, makes sense. There's just the little detail that if the code is moved to the end of `resolveDependencies()` then, the above bits of that function must not throw, because otherwise the clean-up would not be executed. 

Is this something, we intentionally say is not needed?

Reply:
> Is this something, we intentionally say is not needed?

I'd say so, yes. We already have several `resolveDependencies()` implementations that use `try` / `finally` explicitly instead (e.g. [this](https://github.com/oss-review-toolkit/ort/blob/01f19307c486f63012ad8b86c08f323622d87abd/plugins/package-managers/conan/src/main/kotlin/Conan.kt#L147-L154)).